src/main/java/com/rcv/JsonParser.java

```diff
@@ -29,37 +29,43 @@
 
 class JsonParser {
 
-  // function: createRawContestConfigFromFile
-  // purpose: parse input json file into a RawContestConfig
+  // function: parseObjectFromFile
+  // purpose: parse input json file into an object of the specified type
   // param: jsonFilePath path to json file to be parsed into java
+  // param: valueType class of the object to be created from parsed json
   // file access: read
-  // returns: instance of the RawContestConfig parsed from json, or null if there was a problem
-  static RawContestConfig createRawContestConfigFromFile(String jsonFilePath) {
-    RawContestConfig rawConfig;
+  // returns: instance of the object parsed from json or null if there was a problem
+  static <T> T parseObjectFromFile(String jsonFilePath, Class<T> valueType) {
```

Comment:
I'm curious why you bothered to make these methods generic. Will that have a practical purpose in the future?

Reply:
The main motivator here is consolidating the ContestConfig construction code.  Now, originally these functions were generic when I wrote them.  I'm not sure why it was changed to be specific to RawContestConfig, as that conflates json serializing with RawContestConfig construction logic and needlessly removes the generic capability.  Additionally this does have a practical use here.  When I was writing regression testing code, I used JsonParser with a new ContestSummary object to do comparisons.  I did not end up going that route (line by line file comparison proved more useful) but this code helped me explore that option.  I don't have another specific use case in mind atm, but there's no reason not to have this api.